File: scripts/test-extract-tables.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import zipfile
import xml.etree.ElementTree as ET
import sys
import os

def extract_tables_from_docx(file_path):
    """חילוץ טבלאות מקובץ DOCX"""
    try:
        with zipfile.ZipFile(file_path, 'r') as zip_file:
            # קריאת document.xml
            doc_xml = zip_file.read('word/document.xml').decode('utf-8')
            
            # פרסור XML
            root = ET.fromstring(doc_xml)
            
            # חיפוש טבלאות
            namespaces = {
                'w': 'http://schemas.openxmlformats.org/wordprocessingml/2006/main'
            }
            
            tables = root.findall('.//w:tbl', namespaces)
            print(f"נמצאו {len(tables)} טבלאות במסמך\n")
            
            for idx, table in enumerate(tables, 1):
                print(f"=== טבלה {idx} ===")
                rows = table.findall('.//w:tr', namespaces)
                print(f"מספר שורות: {len(rows)}\n")
                
                for row_idx, row in enumerate(rows[:3], 1):  # הדפסת 3 שורות ראשונות
                    cells = row.findall('.//w:tc', namespaces)
                    cell_texts = []
                    for cell in cells:
                        texts = cell.findall('.//w:t', namespaces)
                        cell_text = ' '.join([t.text or '' for t in texts])
                        cell_texts.append(cell_text.strip())
                    print(f"שורה {row_idx}: {' | '.join(cell_texts)}")
                
                if len(rows) > 3:
                    print(f"... ועוד {len(rows) - 3} שורות\n")
                else:
                    print()
                    
    except Exception as e:
        print(f"שגיאה: {e}")
        import traceback
        traceback.print_exc()

if __name__ == '__main__':
    file_path = sys.argv[1] if len(sys.argv) > 1 else r'C:\Scaling Calibration HUB\מאזני לחות.docx'
    if os.path.exists(file_path):
        extract_tables_from_docx(file_path)
    else:
        print(f"קובץ לא נמצא: {file_path}")

File: scripts/test-extract-tables.js
const mammoth = require('mammoth');
const fs = require('fs');
const path = require('path');

const filePath = process.argv[2] || path.join(__dirname, '../../מאזני לחות.docx');

console.log('בודק קובץ:', filePath);

if (!fs.existsSync(filePath)) {
  console.error('קובץ לא נמצא:', filePath);
  process.exit(1);
}

async function testExtractTables() {
  try {
    const buffer = fs.readFileSync(filePath);
    
    // חילוץ HTML
    const htmlResult = await mammoth.convertToHtml({ buffer });
    const html = htmlResult.value;
    
    console.log('\n=== HTML תוצאה (1000 תווים ראשונים) ===');
    console.log(html.substring(0, 1000));
    
    // חילוץ טקסט גולמי
    const textResult = await mammoth.extractRawText({ buffer });
    const text = textResult.value;
    
    console.log('\n=== טקסט גולמי (500 תווים ראשונים) ===');
    console.log(text.substring(0, 500));
    
    // חיפוש טבלאות ב-HTML
    const tableRegex = /<table[^>]*>([\s\S]*?)<\/table>/gi;
    const tables = [];
    let match;
    
    while ((match = tableRegex.exec(html)) !== null) {
      tables.push(match[1]);
    }
    
    console.log(`\n=== נמצאו ${tables.length} טבלאות ===\n`);
    
    tables.forEach((tableHtml, idx) => {
      console.log(`\n--- טבלה ${idx + 1} ---`);
      const rowRegex = /<tr[^>]*>([\s\S]*?)<\/tr>/gi;
      const rows = [];
      let rowMatch;
      
      while ((rowMatch = rowRegex.exec(tableHtml)) !== null) {
        rows.push(rowMatch[1]);
      }
      
      console.log(`מספר שורות: ${rows.length}`);
      
      // הדפסת 3 שורות ראשונות
      rows.slice(0, 3).forEach((rowHtml, rowIdx) => {
        const cellRegex = /<t[dh][^>]*>([\s\S]*?)<\/t[dh]>/gi;
        const cells = [];
        let cellMatch;
        
        while ((cellMatch = cellRegex.exec(rowHtml)) !== null) {
          const cellText = cellMatch[1]
            .replace(/<[^>]+>/g, ' ')
            .replace(/&nbsp;/g, ' ')
            .replace(/\s+/g, ' ')
            .trim();
          cells.push(cellText);
        }
        
        console.log(`שורה ${rowIdx + 1}: ${cells.join(' | ')}`);
      });
    });
    
  } catch (error) {
    console.error('שגיאה:', error);
    process.exit(1);
  }
}

testExtractTables();











